src/main/java/com/gooddata/warehouse/WarehouseS3Credentials.java

```diff
@@ -134,52 +110,36 @@ public String getRegion() {
      * @return S3 credentials with secret key
      */
     public WarehouseS3Credentials withSecretKey(final String secretKey) {
-        return new WarehouseS3Credentials(region, accessKey, secretKey, updatedBy, updated, links);
-    }
-
-    /**
-     * Adds links
-     *
-     * @param links links to add
-     * @return S3 credentials with links
-     */
-    WarehouseS3Credentials withLinks(final Map<String, String> links) {
-        return new WarehouseS3Credentials(region, accessKey, secretKey, updatedBy, updated, links);
+        return new WarehouseS3Credentials(region, accessKey, secretKey, updated, links);
```

Comment:
I suggest to get rid of these "with" methods and make the object mutable, as the other DTOs are.

Reply:
done